vllm/engine/arg_utils.py

```diff
@@ -1454,11 +1454,36 @@ def _set_default_args_v1(self, usage_context: UsageContext) -> None:
             }
             default_max_num_seqs = 256
 
+        # tpu specific default values.
+        if current_platform.is_tpu():
+            default_max_num_batched_tokens_tpu = {
+                UsageContext.LLM_CLASS: {
+                    'v6e': 2048,
+                    'v5e': 1024,
+                    'v5p': 512,
+                },
+                UsageContext.OPENAI_API_SERVER: {
+                    'v6e': 1024,
+                    'v5e': 512,
+                    'v5p': 256,
+                }
+            }
+
         use_context_value = usage_context.value if usage_context else None
         if (self.max_num_batched_tokens is None
                 and usage_context in default_max_num_batched_tokens):
-            self.max_num_batched_tokens = default_max_num_batched_tokens[
-                usage_context]
+            if current_platform.is_tpu():
+                from tpu_info import device
+                try:
+                    chip_type, _ = device.get_local_chips()
+                    self.max_num_batched_tokens = default_max_num_batched_tokens_tpu[
+                        usage_context][chip_type.name.lower()]
+                except Exception:
+                    self.max_num_batched_tokens = default_max_num_batched_tokens[
+                        usage_context]
```

Comment:
It would be better to implement get_device_name() in the TpuPlatform and use that directly rather than inline it here
https://github.com/vllm-project/vllm/blob/200bbf92e8861e2458a6f90bca73f40cc3b1ad1f/vllm/platforms/tpu.py#L55-L57

See CUDA or ROCm for examples 
https://github.com/vllm-project/vllm/blob/200bbf92e8861e2458a6f90bca73f40cc3b1ad1f/vllm/platforms/cuda.py#L347-L351
https://github.com/vllm-project/vllm/blob/200bbf92e8861e2458a6f90bca73f40cc3b1ad1f/vllm/platforms/rocm.py#L219-L225